force-app/main/default/classes/Addresses.cls

```diff
@@ -165,210 +166,293 @@ public inherited sharing class Addresses extends fflib_SObjects {
+    }
+
+    private static Boolean shouldUpdateBillingAddressUndeliverableStatus(NPSP_HouseholdAccount npspHouseholdAccount, NPSP_Address addressToSyncToAccountsAndContacts) {
+        return npspHouseholdAccount.undeliverableAddressStatus() !=
+                addressToSyncToAccountsAndContacts.isUndeliverable();
+    }
+
+    private static Boolean shouldUpdateBillingAddress(NPSP_HouseholdAccount npspHouseholdAccount, NPSP_Address addressToSyncToAccountsAndContacts) {
+        return npspHouseholdAccount.billingAddressIsDifferentFrom(addressToSyncToAccountsAndContacts);
+    }
 
-        if (listAccUpdate.size() > 0) {
-            dmlWrapper.objectsToUpdate.addAll((List<SObject>)listAccUpdate);
+    private static Map<Id, HouseholdInfo> getHouseholdAccountAddressInfoByAccountId(Set<Id> accountIds, Map<Id,
+            Address__c> defaultSeasonalAddressesByAccountId, Map<Id, List<Address__c>> allAddressesByAccountId) {
+        Map<Id, HouseholdInfo> householdInfoByAccountId = new Map<Id, HouseholdInfo>();
+        setAddressForHouseholdInfo(accountIds,
+                defaultSeasonalAddressesByAccountId,
+                allAddressesByAccountId,
+                householdInfoByAccountId);
+
+        return householdInfoByAccountId;
+    }
+
+    private static void setAddressForHouseholdInfo(Set<Id> accountIds, Map<Id, Address__c>
+            defaultSeasonalAddressesByAccountId, Map<Id, List<Address__c>> allAddressesByAccountId, Map<Id, HouseholdInfo> householdInfoByAccountId) {
+        for (Id accountId : accountIds) {
+            // create a HHInfo for each household
+            HouseholdInfo householdInfo = getHouseholdInfoForAccount(accountId);
+
+            // if there is a provided Default address, use it over all other defaults.
+            setHouseholdAddressesToDefaultAddress(defaultSeasonalAddressesByAccountId, accountId, householdInfo, allAddressesByAccountId);
+
+            // finally, if no default found, create an empty default
+            if (householdHasNoDefaultAddress(householdInfo)) {
+                setNewHouseholdDefaultAddress(accountId, householdInfo);
+            }
+            householdInfoByAccountId.put(accountId, householdInfo);
         }
-        if (listAddrUpdate.size() > 0) {
-            dmlWrapper.objectsToUpdate.addAll((List<SObject>)listAddrUpdate);
+    }
+
+    private static void setNewHouseholdDefaultAddress(Id accountId, HouseholdInfo householdInfo) {
+        Address__c defaultAddress = new Address__c();
+        defaultAddress.Household_Account__c = accountId;
+        householdInfo.defaultAddress = defaultAddress;
+    }
+
+    private static Boolean householdHasNoDefaultAddress(HouseholdInfo householdInfo) {
+        return householdInfo.defaultAddress == null;
+    }
+
+    private static void setHouseholdAddressesToDefaultAddress(Map<Id, Address__c> defaultSeasonalAddressesByAccountId, Id accountId, HouseholdInfo householdInfo, Map<Id, List<Address__c>> allAddressesByAccountId) {
```

Comment:
There are some really long lines in this file.  Can we go to multiple lines for those?